crates/holochain/src/conductor/handle.rs

```diff
@@ -894,31 +902,75 @@ impl ConductorHandleT for ConductorHandleImpl {
     async fn create_clone_cell(
         self: Arc<Self>,
         payload: CreateCloneCellPayload,
-    ) -> ConductorResult<CellId> {
+    ) -> ConductorResult<InstalledCell> {
         let CreateCloneCellPayload {
-            properties,
-            dna_hash,
-            installed_app_id,
-            agent_key,
+            app_id,
             role_id,
+            properties,
+            network_seed,
             membrane_proof,
+            origin_time,
+            name,
         } = payload;
-        let cell_id = CellId::new(dna_hash, agent_key);
-        let cells = vec![(cell_id.clone(), membrane_proof)];
-
-        // Run genesis on cells.
-        crate::conductor::conductor::genesis_cells(&self.conductor, cells, self.clone()).await?;
+        if network_seed == None && properties == None {
+            return Err(ConductorError::CloneCellError(
+                "neither network_seed nor properties provided for clone cell".to_string(),
+            ));
+        }
+        let state = self.conductor.get_state().await?;
+        let app = state.get_app(&app_id)?;
+        app.provisioned_cells()
+            .find(|(app_role_id, _)| **app_role_id == role_id)
+            .ok_or_else(|| {
+                ConductorError::CloneCellError(
+                    "no base cell found for provided role id".to_string(),
+                )
+            })?;
+        let role = app.role(&role_id)?;
+        if role.is_clone_limit_reached() {
+            return Err(ConductorError::AppError(AppError::CloneLimitExceeded(
+                role.clone_limit(),
+                role.clone(),
+            )));
+        }
```

Comment:
This check already happens in `add_clone_cell_to_app`, so can it be removed here?